src/main/java/io/confluent/kafkarest/SimpleConsumerManager.java

```diff
@@ -0,0 +1,230 @@
+ **/
+package io.confluent.kafkarest;
+
+import io.confluent.kafka.serializers.KafkaAvroDecoder;
+import io.confluent.kafkarest.converters.AvroConverter;
+import io.confluent.kafkarest.entities.AvroConsumerRecord;
+import io.confluent.kafkarest.entities.BinaryConsumerRecord;
+import io.confluent.kafkarest.entities.ConsumerRecord;
+import io.confluent.kafkarest.entities.EmbeddedFormat;
+import io.confluent.rest.exceptions.RestException;
+import io.confluent.rest.exceptions.RestServerErrorException;
+import kafka.api.PartitionFetchInfo;
+import kafka.cluster.Broker;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.FetchRequest;
+import kafka.javaapi.FetchResponse;
+import kafka.javaapi.message.ByteBufferMessageSet;
+import kafka.message.MessageAndMetadata;
+import kafka.message.MessageAndOffset;
+import kafka.serializer.Decoder;
+import kafka.serializer.DefaultDecoder;
+import kafka.utils.VerifiableProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.core.Response;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class SimpleConsumerManager {
+
+  private static final Logger log = LoggerFactory.getLogger(SimpleConsumerManager.class);
+
+  private final int maxPoolSize;
+  private final MetadataObserver mdObserver;
+  private final SimpleConsumerFactory simpleConsumerFactory;
+
+  private final ConcurrentMap<Broker, SimpleConsumerPool> simpleConsumersPools;
+
+  private AtomicInteger correlationId = new AtomicInteger(0);
+
+  private final Decoder<Object> avroDecoder;
+  private final Decoder<byte[]> binaryDecoder;
+
+  public SimpleConsumerManager(final KafkaRestConfig config,
+                               final MetadataObserver mdObserver,
+                               final SimpleConsumerFactory simpleConsumerFactory) {
+
+    this.mdObserver = mdObserver;
+    this.simpleConsumerFactory = simpleConsumerFactory;
+
+    maxPoolSize = config.getInt(KafkaRestConfig.SIMPLE_CONSUMER_POOL_SIZE_CONFIG);
+    simpleConsumersPools = new ConcurrentHashMap<Broker, SimpleConsumerPool>();
+
+    // Load decoders
+    Properties props = new Properties();
+    props.setProperty("schema.registry.url",
+        config.getString(KafkaRestConfig.SCHEMA_REGISTRY_URL_CONFIG));
+    avroDecoder = new KafkaAvroDecoder(new VerifiableProperties(props));
+
+    binaryDecoder = new DefaultDecoder(new VerifiableProperties());
+  }
+
+  private SimpleConsumerPool createSimpleConsumerPool() {
+    return new SimpleConsumerPool(maxPoolSize, simpleConsumerFactory);
+  }
+
+  private SimpleFetcher getSimpleFetcher(final Broker broker) {
+    // When upgrading to Java 1.8, use simpleConsumersPools.computeIfAbsent() instead
+    SimpleConsumerPool pool = simpleConsumersPools.get(broker);
+    if (pool == null) {
+      pool = simpleConsumersPools.putIfAbsent(broker, createSimpleConsumerPool());
```

Comment:
Oops, looks like tests are failing because putIfAbsent returns the *previous* value, not the new one. So I think we'll need to do this instead:

```
simpleConsumersPools.putIfAbsent(broker, createSimpleConsumerPool());
pool = simpleConsumerPools.get(broker);
```

That has the potential overhead of creating one more simple consumer pool than required, but it'll just get garbage collected. The second `simpleConsumersPools.get(broker)` call is guaranteed to have the valid value due to the previous `putIfAbsent` call.

Assuming the tests pass after that, I think this is ready to merge!